cmd/harmony/main.go

```diff
@@ -229,21 +230,56 @@ func setupInitialAccount() (isLeader bool) {
 	return isLeader
 }
 
-func setupConsensusKey(nodeConfig *nodeconfig.ConfigType) *bls.PublicKey {
-	consensusPriKey, err := blsgen.LoadBlsKeyWithPassPhrase(*blsKeyFile, blsPassphrase)
+func readMultiBlsKeys(nodeConfig *nodeconfig.ConfigType) error {
+	multiKeyFolder := "./hmy/multikeys"
+	fileList, err := ioutil.ReadDir(multiKeyFolder)
 	if err != nil {
-		fmt.Fprintf(os.Stderr, "ERROR when loading bls key, err :%v\n", err)
-		os.Exit(100)
+		return err
 	}
-	pubKey := consensusPriKey.GetPublicKey()
 
-	// Consensus keys are the BLS12-381 keys used to sign consensus messages
-	nodeConfig.ConsensusPriKey, nodeConfig.ConsensusPubKey = consensusPriKey, consensusPriKey.GetPublicKey()
-	if nodeConfig.ConsensusPriKey == nil || nodeConfig.ConsensusPubKey == nil {
-		fmt.Println("error to get consensus keys.")
-		os.Exit(100)
+	for _, fileInfo := range fileList {
+		if strings.HasSuffix(fileInfo.Name(), ".key") {
+			pass := fileInfo.Name()[:len(fileInfo.Name())-4] + ".txt"
+			passphrase, err := utils.GetPassphraseFromSource("file:" + multiKeyFolder + "/" + pass)
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "ERROR when reading passphrase file: %v\n", err)
+				os.Exit(100)
+			}
+			consensusPriKey, err1 := blsgen.LoadBlsKeyWithPassPhrase(multiKeyFolder+"/"+fileInfo.Name(), passphrase)
+			if err1 != nil {
+				return err1
+			}
+			nodeConfig.ConsensusPriKey = nodeconfig.AppendPriKey(nodeConfig.ConsensusPriKey, consensusPriKey)
+			nodeConfig.ConsensusPubKey = nodeconfig.AppendPubKey(nodeConfig.ConsensusPubKey, consensusPriKey.GetPublicKey())
+		}
+	}
+
+	return nil
+}
+
+func setupConsensusKey(nodeConfig *nodeconfig.ConfigType) *bls.PublicKey {
+	if stat, err := os.Stat("./hmy/multikeys"); err == nil && stat.IsDir() {
+		if err := readMultiBlsKeys(nodeConfig); err != nil {
+			fmt.Fprintf(os.Stderr, "ERROR when loading bls key, err :%v\n", err)
+			os.Exit(100)
+		}
+	} else {
+		consensusPriKey, err := blsgen.LoadBlsKeyWithPassPhrase(*blsKeyFile, blsPassphrase)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "ERROR when loading bls key, err :%v\n", err)
+			os.Exit(100)
+		}
+
+		// Consensus keys are the BLS12-381 keys used to sign consensus messages
+		nodeConfig.ConsensusPriKey = nodeconfig.AppendPriKey(nodeConfig.ConsensusPriKey, consensusPriKey)
+		nodeConfig.ConsensusPubKey = nodeconfig.AppendPubKey(nodeConfig.ConsensusPubKey, consensusPriKey.GetPublicKey())
+		if len(nodeConfig.ConsensusPriKey.PrivateKey) == 0 || len(nodeConfig.ConsensusPubKey.PublicKey) == 0 {
+			fmt.Println("error to get consensus keys.")
+			os.Exit(100)
+		}
 	}
-	return pubKey
+
+	return nodeConfig.ConsensusPubKey.PublicKey[0]
```

Comment:
how you know [0] will always exist , what if file dir was empty

Reply:
Added check.